rrmngmnt/power_manager.py

```diff
@@ -0,0 +1,129 @@
+"""
+Manage host power via ssh or ipmitool
+"""
+import socket
+import subprocess
+from rrmngmnt.service import Service
+
+
+class PowerManager(Service):
+    """
+    Base power management class
+    """
+    reboot_command = None
+    poweron_command = None
+    poweroff_command = None
+
+    def _exec_pm_command(self, command, *args):
+        raise NotImplementedError("Must be implemented under child class")
+
+    def restart(self, *args):
+        """
+        Reboot host
+        """
+        self._exec_pm_command(self.reboot_command, *args)
+
+    def poweroff(self, *args):
+        """
+        Power off host
+        """
+        self._exec_pm_command(self.poweroff_command, *args)
+
+    def poweron(self, *args):
+        """
+        Power on host
+        """
+        self._exec_pm_command(self.poweron_command, *args)
+
+
+class SSHPowerManager(PowerManager):
+    """
+    SSH power management class
+    """
+    reboot_command = ["reboot"]
+    poweroff_command = ["poweroff"]
+
+    def _exec_pm_command(self, command, *args):
+        try:
+            command += args
```

Comment:
you are lucky here becasue, args is empty, but in case it is not, that class members will grow ...